spring-cloud-dataflow-shell-core/src/main/java/org/springframework/cloud/dataflow/shell/command/AppRegistryCommands.java

```diff
@@ -248,7 +249,13 @@ public String importFromResource(
 			try {
 				Resource resource = this.resourceLoader.getResource(uri);
 				Properties applications = PropertiesLoaderUtils.loadProperties(resource);
-				PagedResources<AppRegistrationResource> registered = appRegistryOperations().registerAll(applications, force);
+				PagedResources<AppRegistrationResource> registered = null;
+				try {
+					registered = appRegistryOperations().registerAll(applications, force);
+				}
+				catch (Exception e) {
+					return "Error when registering applications from " + uri + ": " + e.getMessage();
```

Comment:
If we want to add a specific `AppRegistrationException` then I think that should live in the spring-cloud-deployer project where `UriRegistry` and `UriRegistryPopulator`are defined. I really don't see a strong need for that though.